### ISSUE TITLE
NOT EXISTS meta query (help request / suggestion / pre-PR query)

### ISSUE THREAD

**Issue by phlbnks** [johnbillion/extended-cpts#117]
I implemented a checkbox for a key meta field not being present by using 

```
'important_missing'      => [
    'meta_exists' => [
        'important' => 'Important missing',
    ],
    'meta_query'  => [
        'compare' => 'IN',
    ],
],
```

but this only return posts where the meta exists but is empty. 

This
```
'important_missing'      => [
    'meta_exists' => [
        'important' => 'Important missing',
    ],
    'meta_query'  => [
        'compare' => 'NOT EXISTS',
    ],
],
```
returns posts where the meta_key doesn't exist.

Is it possible to combine both with the current system? If not, would you accept a PR for a meta_not_exists feature?

Cheers,
Phil

**Comment by ibes** [comment]
https://github.com/johnbillion/extended-cpts/blob/89ba13fa10c66b77f21aacdfb794ef3607814ed5/src/class-extended-cpt.php#L345

https://github.com/johnbillion/extended-cpts/blob/89ba13fa10c66b77f21aacdfb794ef3607814ed5/src/class-extended-cpt.php#L355

Seems like you combine these two.

Tried to give a helpful answer - but not sure if that is helpful :)

**Comment by johnbillion** [comment]
Hi @emirpprime , I shipped a new `meta_key_exists` filter type in 4.2.6. The difference between this and the `meta_exists` filter type is that the new filter doesn't care about the meta field value, it just does an `EXISTS` check.

I can't quite understand what you're asking for based on your description. Does the following do what you want?

```php
'important_missing'      => [
    'meta_key_exists' => [
        'important' => 'Important missing',
    ],
    'meta_query'  => [
        'compare' => 'NOT EXISTS',
    ],
],
```
